bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,580 @@
+    String endpoint = dataSettings.getStubSettings().getEndpoint();
+    return endpoint.substring(0, endpoint.lastIndexOf(":"));
+  }
+
+  @Override
+  public String getAdminHost() {
+    String endpoint = tableAdminSettings.getStubSettings().getEndpoint();
+    return endpoint.substring(0, endpoint.lastIndexOf(":"));
+  }
+
+  @Override
+  public int getPort() {
+    String endpoint = dataSettings.getStubSettings().getEndpoint();
+    return Integer.parseInt(endpoint.substring(endpoint.lastIndexOf(":") + 1));
+  }
+
+  @Override
+  public int getBulkMaxRowCount() {
+    return dataSettings
+        .getStubSettings()
+        .bulkMutateRowsSettings()
+        .getBatchingSettings()
+        .getElementCountThreshold()
+        .intValue();
+  }
+
+  // ************** Getters **************
+  public boolean isChannelPoolCachingEnabled() {
+    // This is primarily used by Dataflow where connections open and close often. This is a
+    // performance optimization that will reduce the cost to open connections.
+    return configuration.getBoolean(BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL, false);
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableDataSettings}. */
+  public BigtableDataSettings getDataSettings() {
+    return dataSettings;
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableTableAdminSettings}. */
+  public BigtableTableAdminSettings getTableAdminSettings() {
+    return tableAdminSettings;
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableInstanceAdminSettings}. */
+  @Nullable
+  public BigtableInstanceAdminSettings getInstanceAdminSettings() {
+    return instanceAdminSettings;
+  }
+
+  // ************** Private Helpers **************
+  private BigtableDataSettings buildBigtableDataSettings() throws IOException {
+    BigtableDataSettings.Builder dataBuilder =
+        BigtableDataSettings.newBuilder()
+            .setProjectId(getProjectId())
+            .setInstanceId(getInstanceId());
+
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+    if (!isNullOrEmpty(appProfileId)) {
+      dataBuilder.setAppProfileId(appProfileId);
+    }
+
+    EnhancedBigtableStubSettings.Builder stubSettings = dataBuilder.stubSettings();
+
+    buildEndpoints(stubSettings, BIGTABLE_HOST_KEY);
+
+    buildHeaderProvider(stubSettings);
+
+    buildCredentialProvider(stubSettings);
+
+    // RPC methods
+    buildBulkMutationsSettings(stubSettings);
+
+    buildBulkReadRowsSettings(stubSettings);
+
+    buildReadRowsSettings(stubSettings);
+
+    buildNonIdempotentCallSettings(stubSettings.checkAndMutateRowSettings());
+    buildNonIdempotentCallSettings(stubSettings.readModifyWriteRowSettings());
+
+    buildIdempotentCallSettings(stubSettings.readRowSettings());
+    buildIdempotentCallSettings(stubSettings.mutateRowSettings());
+    buildIdempotentCallSettings(stubSettings.sampleRowKeysSettings());
+
+    buildEmulatorSettings(stubSettings);
```

Comment:
can we move to line 192 after credential configuration...it affects similar things as credentials & endpoints